lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -183,44 +195,59 @@ class ConvertAtenAddSubOp : public OpConversionPattern<AtenOpT> {
   matchAndRewrite(AtenOpT op, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
     Value lhs = adaptor.self();
-    auto lhsTy = lhs.getType().dyn_cast<TensorType>();
+    auto lhsType = lhs.getType().dyn_cast<TensorType>();
     Value rhs = adaptor.other();
-    auto rhsTy = rhs.getType().dyn_cast<TensorType>();
+    auto rhsType = rhs.getType().dyn_cast<TensorType>();
 
-    if (!lhsTy)
+    if (!lhsType)
       return op.emitError("Only Tensor types supported in TOSA");
 
-    auto lhsElemTy = lhsTy.getElementType();
-    if (!lhsElemTy.isIntOrFloat())
+    if (auto lhsElemTy = lhsType.getElementType().dyn_cast<IntegerType>()) {
+      if (lhsElemTy.getWidth() > 32)
+        return op.emitError(
+            "Integers with widths greater than 32 are not supported");
+    }
+
+    auto outType =
+        static_cast<mlir::Type>(
+            OpConversionPattern<AtenOpT>::getTypeConverter()->convertType(
+                op.getType()))
+            .cast<TensorType>();
+
+    Type outElemTy = outType.getElementType();
+    if (!outElemTy.isIntOrFloat()) {
       return op.emitError(
           "Only floating-point or integer datatype legalization supported");
+    }
 
     Value rhsAsTensor;
-    if (!rhsTy) {
+    if (!rhsType) {
       if (failed(torchScalarToTosaTensor(rewriter, op.getOperation(),
```

Comment:
nit: we can pass `op` without using `getOperation`. Could you also fix other places in this file as well.